python_modules/dagster/dagster/_core/types/connection.py

```diff
@@ -83,3 +83,30 @@ def from_cursor(cls, cursor: str):
         if "value" not in raw:
             raise ValueError(f"Invalid cursor: {cursor}")
         return ValueIndexCursor(value=raw["value"])
+
+
+@record
+class StorageIdCursor:
+    storage_id: int
```

Comment:
we should probably start using this in many places where we currently pass around naked ints